storm-core/src/jvm/org/apache/storm/scheduler/blacklist/BlacklistScheduler.java

```diff
@@ -0,0 +1,252 @@
+    private IBlacklistStrategy blacklistStrategy;
+
+    private int nimbusMonitorFreqSecs;
+
+    private Map<String, Set<Integer>> cachedSupervisors;
+
+    //key is supervisor key ,value is supervisor ports
+    private CircularBuffer<HashMap<String, Set<Integer>>> badSupervisorsTolerance;
+    private Set<String> blacklistHost;
+
+    public BlacklistScheduler(IScheduler underlyingScheduler) {
+        this.underlyingScheduler = underlyingScheduler;
+    }
+
+    @Override
+    public void prepare(Map conf) {
+        LOG.info("prepare black list scheduler");
+        LOG.info(conf.toString());
+        underlyingScheduler.prepare(conf);
+        _conf = conf;
+        if (_conf.containsKey(Config.BLACKLIST_SCHEDULER_TOLERANCE_TIME)) {
+            toleranceTime = (Integer) _conf.get(Config.BLACKLIST_SCHEDULER_TOLERANCE_TIME);
+        }
+        if (_conf.containsKey(Config.BLACKLIST_SCHEDULER_TOLERANCE_COUNT)) {
+            toleranceCount = (Integer) _conf.get(Config.BLACKLIST_SCHEDULER_TOLERANCE_COUNT);
+        }
+        if (_conf.containsKey(Config.BLACKLIST_SCHEDULER_RESUME_TIME)) {
+            resumeTime = (Integer) _conf.get(Config.BLACKLIST_SCHEDULER_RESUME_TIME);
+        }
+        String reporterClassName = _conf.containsKey(Config.BLACKLIST_SCHEDULER_REPORTER) ? (String) _conf.get(Config.BLACKLIST_SCHEDULER_REPORTER) : "";
+        try {
+            reporter = (IReporter) Class.forName(reporterClassName).newInstance();
+        } catch (ClassNotFoundException e) {
+            LOG.error("Can't find blacklist reporter for name {}", reporterClassName);
+            throw new RuntimeException(e);
+        } catch (InstantiationException e) {
+            LOG.error("Throw InstantiationException blacklist reporter for name {}", reporterClassName);
+            throw new RuntimeException(e);
+        } catch (IllegalAccessException e) {
+            LOG.error("Throw illegalAccessException blacklist reporter for name {}", reporterClassName);
+            throw new RuntimeException(e);
+        }
+
+        String strategyClassName = _conf.containsKey(Config.BLACKLIST_SCHEDULER_STRATEGY) ? (String) _conf.get(Config.BLACKLIST_SCHEDULER_STRATEGY) : "";
+        try {
+            blacklistStrategy = (IBlacklistStrategy) Class.forName(strategyClassName).newInstance();
+        } catch (ClassNotFoundException e) {
+            LOG.error("Can't find blacklist strategy for name {}", strategyClassName);
+            throw new RuntimeException(e);
+        } catch (InstantiationException e) {
+            LOG.error("Throw InstantiationException blacklist strategy for name {}", strategyClassName);
+            throw new RuntimeException(e);
+        } catch (IllegalAccessException e) {
+            LOG.error("Throw illegalAccessException blacklist strategy for name {}", strategyClassName);
+            throw new RuntimeException(e);
+        }
+
+        nimbusMonitorFreqSecs = (Integer) _conf.get(Config.NIMBUS_MONITOR_FREQ_SECS);
```

Comment:
I think it's hard to avoid the influence of NIMBUS_MONITOR_FREQ_SECS, as the blacklist scheduler is a SCHEDULER at first. It is called every NIMBUS_MONITOR_FREQ_SECS.
I take this conf here only to calculate the length of sliding window. The window size is "toleranceTime / nimbusMonitorFreqSecs ", so every time the scheduler is called ,the window will slide for 1 step.
I think even if I create a new configure for this, it MUST equal to NIMBUS_MONITOR_FREQ_SECS so that the sliding window can work properly . So there is no need to do so.